pdr_backend/util/currency_types.py

```diff
@@ -1,147 +1,144 @@
 import logging
+from abc import ABC
 from typing import Union
 from enforce_typing import enforce_types
 
 logger = logging.getLogger("currency_types")
 
 
 @enforce_types
-class Eth:
-    def __init__(self, amt_eth: Union[int, float]):
-        if amt_eth > 100_000_000_000:
-            logger.warning(
-                "amt_eth=%s is very large. Should it be wei instead?", amt_eth
-            )
-
-        self.amt_eth = amt_eth
-
-    def to_wei(self) -> "Wei":
-        return Wei(int(self.amt_eth * 1e18))
+class EthUnit(ABC):
+    def __init__(self, amount: Union[int, float]):
+        self.amount = amount
 
     def __str__(self) -> str:
-        return f"{self.amt_eth} eth"
+        return f"{self.amount}"
```

Comment:
I would keep the quantifier to differentiate that it is not a regular number, when printing. But I'm fine either way.